### PR TITLE
Lotsa stuff

### PR DESCRIPTION
**Pull request by terrycain**
@JackDanger sorry in advance, is quite a bit to review :wink:. Got a bit bored and had little access to internet so went on a service completion spree.

### CODE REVIEW

**Comment by terrycain** [comment]
I'd suggest reviewing commit by commit as I tried to group the changes logically

**Comment by JackDanger** [comment]
This is awesome. You're a machine.

@toshitanian or @captainkerk – would either (or both) of you like to do a review of this PR that @terrycain put together?

**Comment by JackDanger** [comment]
(I'm traveling and will be in Morocco for a day or so and off the internet)

**Comment by captainkerk** [comment]
Sure can. I'll get back to this later this afternoon.

File: moto/dynamodb2/models.py [hunk 1]
```diff
@@ -5,9 +5,11 @@
 import json
 import re
 
+import boto3
```
Comment (captainkerk) [inline moto/dynamodb2/models.py]: nix this?

Comment (terrycain) [reply moto/dynamodb2/models.py]: Is used to grab a list of regions that dynamo is present in, I'd rather that than hardcoding regions. Is also part of the script that makes models.py templates.

Comment (captainkerk) [reply moto/dynamodb2/models.py]: Ohh I see! You also did that in the batch and polly services. This makes sense now.

**Comment by captainkerk** [comment]
Sorry—I'm happy to help, but i've had very little free time over the past few days. I'll try to really get back to this tomorrow afternoon.

**Comment by terrycain** [comment]
No problem, am not in any rush with this :smile:

**Comment by spulec** [comment]
This is amazing @terrycain. Thanks!

**Pull request merged**